src/pydicom/pixels/encoders/base.py

```diff
@@ -18,113 +26,472 @@
+        return out
+
+    def set_encoders(self, encoders: dict[str, EncodeFunction]) -> None:
+        """Set the encoders use for encoding compressed pixel data.
+
+        Parameters
+        ----------
+        encoders : dict[str, EncodeFunction]
+            A dict of {name: encoder function}.
+        """
+        self._encoders = encoders
+
+    def set_source(self, src: "np.ndarray | Dataset | Buffer") -> None:
+        """Set the pixel data to be encoded.
+
+        Parameters
+        ----------
+        src : bytes | bytearray | memoryview | pydicom.dataset.Dataset | numpy.ndarray
+
+            * If a buffer-like then the encoded pixel data
+            * If a :class:`~pydicom.dataset.Dataset` then a dataset containing
+              the pixel data and associated group ``0x0028`` elements.
+            * If a :class:`numpy.ndarray` then an array containing the image data.
+        """
+        if isinstance(src, Dataset):
+            self._set_options_ds(src)
+            self._src = src.PixelData
+            self._src_type = "Dataset"
+        elif isinstance(src, (bytes | bytearray | memoryview)):
+            self._src = src
+            self._src_type = "Buffer"
+        elif isinstance(src, np.ndarray):
+            # Ensure the array is in the required byte order (little-endian)
+            sys_endianness = "<" if sys.byteorder == "little" else ">"
+            # `byteorder` may be
+            #   '|': none available, such as for 8 bit -> ignore
+            #   '=': native system endianness -> change to '<' or '>'
+            #   '<' or '>': little or big
+            byteorder = src.dtype.byteorder
+            byteorder = sys_endianness if byteorder == "=" else byteorder
+            if byteorder == ">":
+                src = src.astype(src.dtype.newbyteorder("<"))
+
+            self._src = src
+            self._src_type = "Array"
+        else:
+            raise TypeError(
+                "'src' must be bytes, numpy.ndarray or pydicom.dataset.Dataset, "
+                f"not '{src.__class__.__name__}'"
+            )
+
+    @property
+    def src(self) -> "Buffer | np.ndarray":
+        """Return the buffer-like or :class:`numpy.ndarray` containing the pixel data."""
+        return self._src
+
+    def __str__(self) -> str:
+        """Return nice string output for the runner."""
+        s = [f"EncodeRunner for '{self.transfer_syntax.name}'"]
+        s.append("Options")
+        s.extend([f"  {name}: {value}" for name, value in self.options.items()])
+        if self._encoders:
+            s.append("Encoders")
+            s.extend([f"  {name}" for name in self._encoders])
+
+        return "\n".join(s)
+
+    def validate(self) -> None:
+        """Validate the encoding options and source pixel data."""
+        self._validate_options()
+        if self.is_dataset or self.is_buffer:
+            self._validate_buffer()
+            if self.is_buffer:
+                return
```

Comment:
`self.is_buffer` is treated as advanced usage so the encoding profile check is skipped.

Although you could just pass `validate=False` instead? Maybe include it then?